wordpress-mobile/WordPress-Android
# Add entry point for Design System

Part of issue #19802 

This PR adds an entry point into the Design System under the `Me` tab for `debug` builds only. When you click on this setting, it should take you to a screen with 2 buttons: `Foundation` and `Components`. This PR also adds the [Compose Navigation dependency](https://developer.android.com/jetpack/compose/navigation).

Note: The screen below is a starter screen that is a work in progress. Its functionality/theming will be updated in future PRs. Future PRs will also add a custom Compose theme, which will include specifications for things like color schemes, typography, buttons, and more. 

## To Test:
1. Navigate to the `Me` tab. If you're on a debug build, you should see the `Design System` row. If on a release version, you should not see the row.
2. Clicking on the `Design System` row should take you to the 2nd screen in the chart below.

| `Me` tab with entry point  | Starter screen after clicking Design System row |
|-----------|--------------------------------------|
|       <img width="408" alt="Screenshot 2024-01-18 at 6 42 05 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/86ef5c61-a040-4613-80d8-3aebc76baa5a">    |     <img width="411" alt="Screenshot 2024-01-18 at 6 27 58 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/24b22dea-c9ca-4050-8c2b-638f65552492"> |





-----

## Regression Notes

1. Potential unintended areas of impact

    - None

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    - None

4. What automated tests I added (or what prevented me from doing so)

    - None

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
